src/v/archival/segment_reupload.cc

```diff
@@ -176,11 +176,17 @@ void segment_collector::do_collect(segment_collector_mode mode) {
           _segments.empty()
```

Comment:
>archival: always move reupload start forward

I take it this commit is the same as this PR but we'll track the separate PR for backport purposes?
https://github.com/redpanda-data/redpanda/pull/14911

Reply:
That's right. I'm happy to merge that one, and rebase this one